### PR TITLE
Fixed Switch to Default Editor From Symbol Outline

### LINKED ISSUE #124122: Markdown File Outline Links Switching to Static Preview if Static Preview is Default
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.56
- OS Version: Windows

Related to #84520

Steps to Reproduce:

1. Set *.md files to open by default using static preview custom editor.
2. Navigate to text editor for a *.md file and click on any header in the `outline` menu on the bottom left.
3. This should bring the window to the top of the static editor, which is not expected.

### PR DESCRIPTION
**Pull request by andreamah**
This PR fixes #124122

Previously, clicking on symbols in the file outline when you had a different default editor would open your default. This editor override disable should fix it.

### CODE REVIEW

File: src/vs/workbench/contrib/outline/browser/outlinePane.ts [hunk 1]
```diff
@@ -284,7 +285,7 @@ export class OutlinePane extends ViewPane {
 
 		// feature: reveal outline selection in editor
 		// on change -> reveal/select defining range
-		this._editorDisposables.add(tree.onDidOpen(e => newOutline.reveal(e.element, e.editorOptions, e.sideBySide)));
+		this._editorDisposables.add(tree.onDidOpen(e => newOutline.reveal(e.element, { ...e.editorOptions, ...{ override: EditorOverride.DISABLED } }, e.sideBySide)));
```
Comment (mjbvz) [inline src/vs/workbench/contrib/outline/browser/outlinePane.ts:288]:
```suggestion
		this._editorDisposables.add(tree.onDidOpen(e => newOutline.reveal(e.element, { ...e.editorOptions, override: EditorOverride.DISABLED }, e.sideBySide)));
```

**Review by mjbvz — APPROVED** [review]

**Review by lramos15 — APPROVED** [review]
@andreamah Do you mind testing to see how your changes work if a custom editor provides outline information. For example set markdown notebooks to be your default and then test the outline. That's the only case I think might not work with this change, but if it does then I think the changes look good.

**Comment by andreamah** [comment]
> @andreamah Do you mind testing to see how your changes work if a custom editor provides outline information. For example set markdown notebooks to be your default and then test the outline. That's the only case I think might not work with this change, but if it does then I think the changes look good.

Yeah, looks like that doesn't work with the change :/ I'll take a look at how I can change it so that it can only override to the same editor for the symbol reveal.

**Comment by andreamah** [comment]
@lramos15 can you double-check these changes before I merge? Should just be the code that you suggested I use.

**Comment by lramos15** [comment]
> @lramos15 can you double-check these changes before I merge? Should just be the code that you suggested I use.

Did it fix the bug I described above?

**Comment by lramos15** [comment]
👍 Should be good to go then

**Pull request merged**